### PR TITLE
Issue 25 model module

### PR DESCRIPTION
**Pull request by jhamacher-indeed**

### CODE REVIEW

File: rabbitmq-operator-model/build.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
+plugins {
+    id 'java'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
```
Comment (bfink13) [inline rabbitmq-operator-model/build.gradle:9]: i believe a whole bunch of these aren't needed

**Review by bfink13 — CHANGES_REQUESTED** [review]
I think we only want the CRD models to move, right?

**Comment by jhamacher-indeed** [comment]
Some of the files in model/crd/ depend on files in model/rabbitmq/.  For example:
```
package com.indeed.operators.rabbitmq.model.crd.rabbitmq;
...
import com.indeed.operators.rabbitmq.model.rabbitmq.VhostOperationPermissions;
...
public class VhostPermissions {
...
```

I didn't check extensively so maybe there are some non-CRD files that aren't necessary, but it seemed cleaner to move them all.

**Comment by bfink13** [comment]
> Some of the files in model/crd/ depend on files in model/rabbitmq/. For example:

Oof, well that is extremely lame. I checked and it looks like VhostPermissions is the only file in the rabbitmq package that is referenced from the crd package. Can you just move VhostPermissions into the crd package instead? I'd rather not expose more than we have to in the public library.

File: rabbitmq-operator-model/build.gradle (hunk 1 above)
Comment (jhamacher-indeed) [reply rabbitmq-operator-model/build.gradle:9]: I thought I'd verified that all of them were necessary, but it appears that I can remove jackson-databind and the kubernetes-client annotation processor.

I'll check again after moving some of the files out of the model module.

File: rabbitmq-operator-model/build.gradle [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    compile('io.fabric8:kubernetes-client:4.1.3')
+    compile('com.google.guava:guava:27.0.1-jre')
+    compile('com.fasterxml.jackson.core:jackson-annotations:2.9.7')
+
+    compileOnly('io.sundr:builder-annotations:0.14.7')
+
+    annotationProcessor('io.sundr:builder-annotations:0.14.7')
+}
+
+test {
```
Comment (bfink13) [inline rabbitmq-operator-model/build.gradle]: this can be removed

File: rabbitmq-operator/build.gradle [hunk 3]
```diff
@@ -12,24 +12,20 @@ repositories {
...
     compile('org.apache.commons:commons-text:1.6')
-    compile('com.fasterxml.jackson.core:jackson-annotations:2.9.7')
     compile('com.fasterxml.jackson.core:jackson-core:2.9.7')
     compile('com.fasterxml.jackson.core:jackson-databind:2.9.7')
     compile('org.springframework.boot:spring-boot-starter:2.1.1.RELEASE') {
         exclude group: 'org.springframework.boot', module: 'spring-boot-starter-logging'
     }
     compile('org.springframework.boot:spring-boot-starter-log4j2:2.1.1.RELEASE')
 
-    compileOnly('io.sundr:builder-annotations:0.14.7')
```
Comment (bfink13) [inline rabbitmq-operator/build.gradle:28]: nice

File: release.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-docker build -t indeedoss/rabbitmq-operator:latest .
+docker build -t indeedoss/rabbitmq-operator:latest rabbitmq-operator
```
Comment (bfink13) [inline release.sh:4]: the Dockerfile should probably move into the rabbitmq-operator directory since that's all it builds against

Comment (jhamacher-indeed) [reply release.sh:4]: I did move the Dockerfile, back in 3e33903.  I just missed updating this until I saw the Travis test build fail.

Comment (jhamacher-indeed) [reply release.sh:4]: Or did you mean that this file should be moved into the rabbitmq-operator directory as well?  That would require changes to the Travis configuration (I think).

**Comment by jhamacher-indeed** [comment]
@bfink13 any more feedback here?  I think I addressed everything.

**Review by bfink13 — APPROVED** [review]

**Pull request merged**